contracts/root/RootChainManager/RootChainManager.sol

```diff
@@ -165,6 +165,48 @@ contract RootChainManager is
         address childToken,
         bytes32 tokenType
     ) external override only(MAPPER_ROLE) {
+        // explicit check if token is already mapped to avoid accidental remaps
+        require(
+            rootToChildToken[rootToken] == address(0) &&
+            childToRootToken[childToken] == address(0),
+            "RootChainManager: ALREADY_MAPPED"
+        );
+        _mapToken(rootToken, childToken, tokenType);
+    }
+
+    /**
+     * @notice Remap a token that has already been mapped, properly cleans up old mapping
+     * Callable only by mappers
+     * @param rootToken address of token on root chain
+     * @param childToken address of token on child chain
+     * @param tokenType bytes32 unique identifier for the token type
+     */
+    function remapToken(
+        address rootToken,
+        address childToken,
+        bytes32 tokenType
+    ) external override only(MAPPER_ROLE) {
+        // cleanup old mapping
+        address oldChildToken = rootToChildToken[rootToken];
+        address oldRootToken = childToRootToken[childToken];
+
+        if (rootToChildToken[oldRootToken] != address(0)) {
+            rootToChildToken[oldRootToken] = address(0);
+            tokenToType[oldRootToken] = bytes32(0);
+        }
+
+        if (childToRootToken[oldChildToken] != address(0)) {
+            childToRootToken[oldChildToken] = address(0);
+        }
```

Comment:
This is duplicate of `_mapToken`. `_mapToken` doesn't have `tokenToType[oldRootToken] = bytes32(0);`

Reply:
Are you talking about `_mapToken` in ChildChainManager? ChildChainManager doesn't store token type so it is not there.

Also, ChildChainManager doesn't have separate `remap` function because I am assuming we will never map tokens by directly calling it. 

RootChainManager has 2 external functions `map`(with extra revert condition) and `remap`(with extra cleanup logic). Both these call the same private `_map` function.